driver/src/contracts/stablex_auction_element.rs

```diff
@@ -68,19 +68,33 @@ impl StableXAuctionElement {
 fn compute_buy_sell_amounts(
     numerator: u128,
     denominator: u128,
-    amount: u128,
+    remaining: u128,
     is_sell_order: bool,
 ) -> (u128, u128) {
-    if denominator > 0 {
-        let other = (numerator * amount) / denominator;
-        if is_sell_order {
-            (other, amount)
-        } else {
-            (amount, other)
+    // TODO - use u128_to_u256 for overflow handling (this is a conversion used in naive_solver)
+    //    assert!(remaining <= denominator);
+    let mut other = 0;
+    let (buy_amount, sell_amount) = if is_sell_order {
+        // 0 on sellAmount (remaining <= denominator) is nonsense, but solver can handle it.
+        // 0 on buyAmount (numerator) is a Market Sell Order.
+        if denominator > 0 {
+            let p = remaining * numerator;
+            let q = denominator;
+            // Recall that ceil(p / float(q)) == (p + q - 1) / q
```

Comment:
Good call @bh2smith. Nice! I completely missed that the rounding errors could be a significant issue

Reply:
Why exactly do we need the ceil-rounding here? I don't think we do any of this in the smart contract.

Reply:
If you really want to do the rounding here (it makes at most 1 wei difference and I don't think we do it in the smart contract which should be the single source of truth), let's at least share the logic with the naive solver by moving this into a helper method (and making sure this comment has to only be in the codebase once).

Reply:
Yes, just a moment!

Reply:
I made a Branch with this ceiling quotient as a utility function. It still needs some work. If we merge this the way it is, I could continue and make the appropriate changes. Just wanna make sure this is not blocking anyone.

Reply:
I made a Branch with this ceiling quotient as a utility function. It still needs some work. If we merge this the way it is, I could continue and make the appropriate changes. Just wanna make sure this is not blocking anyone.